ee/management/commands/sync_replicated_schema.py

```diff
@@ -0,0 +1,65 @@
+import re
+from typing import Dict, Set
+
+import structlog
+from django.core.management.base import BaseCommand
+
+from ee.clickhouse.client import sync_execute
+from ee.clickhouse.sql.schema import CREATE_TABLE_QUERIES, build_query, get_table_name
+from posthog.settings import CLICKHOUSE_CLUSTER, CLICKHOUSE_DATABASE, CLICKHOUSE_REPLICATION, MULTI_TENANCY
+
+logger = structlog.get_logger(__name__)
+
+
+class Command(BaseCommand):
+    help = "Synchronize schema across clickhouse cluster, creating missing tables on new nodes"
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--dry-run", action="store_true", help="Exits with a non-zero status if schema changes would be required."
+        )
+
+    def handle(self, *args, **options):
+        if not CLICKHOUSE_REPLICATION or MULTI_TENANCY:
+            logger.info("✅ Skipping non-replicated or cloud setup")
+            return
+
+        out_of_sync_hosts = self.get_out_of_sync_hosts()
+
+        if len(out_of_sync_hosts) > 0:
+            logger.info("Schema out of sync on some clickhouse nodes!", out_of_sync_hosts=out_of_sync_hosts)
+
+            if options["dry_run"]:
+                exit(1)
+
+            logger.info("Creating missing tables")
+            for query in CREATE_TABLE_QUERIES:
+                sync_execute(build_query(query))
```

Comment:
I assume these are ON CLUSTER queries, and that will not cause issues if tables are already defined on a node?

Reply:
Exactly - you can also check `schema.py` for the queries.